torch/_dynamo/utils.py

```diff
@@ -787,11 +791,11 @@ def dict_const_keys(value):
     return {k for k in value.keys() if not isinstance(k, torch.nn.Parameter)}
 
 
-def dict_const_keys_repr(const_keys):
+def dict_const_keys_repr(const_keys, guard_source):
     if any(isinstance(k, enum.Enum) for k in const_keys):
         # To workaround repr(Enum) returning invalid global reference before python 3.11
         # by calling enum_repr and removing quotes to render enum in guard code.
-        const_keys_str = f"{ {enum_repr(k) if isinstance(k, enum.Enum) else repr(k) for k in const_keys} }".replace(
+        const_keys_str = f"{ {enum_repr(k, guard_source) if isinstance(k, enum.Enum) else repr(k) for k in const_keys} }".replace(
```

Comment:
This still doesn't make any sense. You're feeding in guard_source to make the decision whether or not to look for the enum namespace from globals or locals, but there isn't necessarily any relationship between the guard source. Let's imagine we're doing a key check on `foobar`. This will look like `___dict_const_keys(foobar) == {EnumNs.blah}`. `guard_source` being local or global says something about `foobar`... but you're using that to decide if you will generate `L["EnumNs"]` or `G["EnumNs"]`

Just unconditionally get this from the global namespace and file a TODO to fix it properly...